packages/stream-model-handler/src/interfaces-utils.ts

```diff
@@ -687,3 +690,11 @@ export async function validateImplementedInterfaces(
     throw new AggregateError(errors, `Interfaces validation failed for model ${model.name}`)
   }
 }
+function isValidImmutabilityImplementation(
+  expected: ModelDefinition,
+  implemented: ModelDefinition
+): boolean {
+  if (implemented.version == '1.0') return true
+  const implementedSet = new Set(implemented.immutableFields)
+  return (expected as ModelDefinitionV2).immutableFields.every((item) => implementedSet.has(item))
```

Comment:
Would be good to add a type-check for the model definition here, for example we could have a v3 of the model definition in the future that changes the format of the `immutableFields`, but the type cast would hide the error here.

Also, I think `immutableFields` can be `undefined` in the definition, no? If so, we need to check it is set.

Reply:
I updated this part, let me know if you think we would need something more robust.